Sources/GRPCCodeGen/Internal/Translator/ServerCodeTranslator.swift

```diff
@@ -0,0 +1,508 @@
+      body: [response, returnStatement]
+    )
+  }
+
+  private func makeResponse(
+    for method: CodeGenerationRequest.ServiceDescriptor.MethodDescriptor
+  ) -> Declaration {
+    let serverRequest: Expression
+    if !method.isInputStreaming {
+      // Transform the streaming request into a unary request.
+      serverRequest = Expression.functionCall(
+        calledExpression: .memberAccess(
+          MemberAccessDescription(
+            left: .identifierPattern("ServerRequest"),
+            right: "Single"
+          )
+        ),
+        arguments: [
+          FunctionArgumentDescription(label: "stream", expression: .identifierPattern("request"))
+        ]
+      )
+    } else {
+      serverRequest = Expression.identifierPattern("request")
+    }
+    // Call to the corresponding ServiceProtocol method.
+    let serviceProtocolMethod = Expression.functionCall(
+      calledExpression: .memberAccess(
+        MemberAccessDescription(left: .identifierPattern("self"), right: method.name)
+      ),
+      arguments: [FunctionArgumentDescription(label: "request", expression: serverRequest)]
+    )
+
+    let responseValue = Expression.unaryKeyword(
+      kind: .try,
+      expression: .unaryKeyword(kind: .await, expression: serviceProtocolMethod)
+    )
+
+    return .variable(kind: .let, left: "response", right: responseValue)
+  }
+
+  private func makeReturnStatement(
+    for method: CodeGenerationRequest.ServiceDescriptor.MethodDescriptor
+  ) -> Expression {
+    let returnValue: Expression
+    // Transforming the unary response into a streaming one.
+    if !method.isOutputStreaming {
+      returnValue = .functionCall(
+        calledExpression: .memberAccess(
+          MemberAccessDescription(
+            left: .identifierType(.member(["ServerResponse"])),
+            right: "Stream"
+          )
+        ),
+        arguments: [
+          (FunctionArgumentDescription(label: "single", expression: .identifierPattern("response")))
+        ]
+      )
+    } else {
+      returnValue = .identifierPattern("response")
+    }
+
+    return .unaryKeyword(kind: .return, expression: returnValue)
+  }
+
+  /// Generates the fully qualified name of the type alias for a service descriptor.
+  private func serviceProtocolName(
+    for service: CodeGenerationRequest.ServiceDescriptor,
+    streaming: Bool
+  ) -> String {
+    let namespacedPrefix: String
+    if service.namespace.isEmpty {
+      namespacedPrefix = service.name
+    } else {
+      namespacedPrefix = "\(service.namespace).\(service.name)"
+    }
+    if streaming {
+      return "\(namespacedPrefix).StreamingServiceProtocol"
+    }
+    return "\(namespacedPrefix).ServiceProtocol"
+  }
+
+  private func fullyQualifiedMethodDescriptor() {}
+
+  fileprivate enum InputOutputType {
+    case input
+    case output
+  }
+
+  /// Generates the fully qualified name of the typealias for the input or output type of a method.
```

Comment:
Not sure here if the comments should be marked like this or with "//" in this case.